### ISSUE TITLE
XmlHTTPrequest Errors

### ISSUE THREAD

**Issue by fr500** [jvilk/BrowserFS#157]
Hi
I've been trying to use XmlHTTPRequest together with a mountable filesystem on an emscripten port of RetroArch (I'm using dropbox for user data and I intend to us XmlHTTPRequest for assets)

I can see the file lists properly but whenever I try to open a file I get these:
http://pastebin.com/n4DPeXtM

Here's the relevant portion of my code:
```
         /* create a mountable filesystem that will server as a root 
            mountpoint for browserfs */
         var mfs =  new BrowserFS.FileSystem.MountableFileSystem();
         /* create a local filesystem */
         var lsfs = new BrowserFS.FileSystem.LocalStorage();
         /* create an XmlHttpRequest filesystem for assets */
         var xfs1 =  new BrowserFS.FileSystem.XmlHttpRequest
            (".index-xhr", "https://bot.libretro.com/web/assets/");
         var xfs2 =  new BrowserFS.FileSystem.XmlHttpRequest
            (".index-xhr", "https://bot.libretro.com/assets/cores/");
         
         /* mount the local filesystem at the root of mfs*/
         mfs.mount('/home/web_user/userdata', lsfs);
         mfs.mount('/home/web_user/democontent', xfs2);
         mfs.mount('/home/web_user/retroarch/bundle', xfs1);

         BrowserFS.initialize(mfs);
         var BFS = new BrowserFS.EmscriptenFS();
         FS.mount(BFS, {root: '/home'}, '/home');
         console.log('WEBPLAYER: Filesystem initialized');
```
Am I doing something wrong?
At the beginning I tried to treat the xmlhttp filesystem as an asyncMirror too but it doesn't work since it expects a mountable filesystem.

**Comment by jvilk** [comment]
I believe you've found a bug! The code on that path uses writeUint8 to automatically clamp values, but does not turn off Buffer's value checking -- which I newly added recently-ish. I will make a bugfix release later on tonight.

A workaround would be to directly download the JSON file, and pass the JSON.parse'd result to the XHR constructor.

**Comment by jvilk** [comment]
Also, AsyncMirror should work with XHRFS?

**Comment by fr500** [comment]
No, it told me that an async file system is expected

**Comment by jvilk** [comment]
Ahhh, right, because the XHRFS supports synchronous operations. I should lift that restriction in case people want to avoid synchronous XHR.

**Comment by jvilk** [comment]
@fr500 [Can you try this build of BrowserFS, and let me know if it works for you?](https://github.com/jvilk/BrowserFS/files/465007/browserfs.zip) I lifted the AsyncMirror restriction, and I believe I fixed the bug.

If all seems well, I'll push the fix and officially release the change.

**Comment by fr500** [comment]
Sure, I'll test tonight

**Comment by fr500** [comment]
Thanks a lot! all is in good working order now and this gives me two options, I can ask the user if he wants to pre-load assets or load on demand

**Comment by jvilk** [comment]
Awesome! I released this in [v0.6.1](https://github.com/jvilk/BrowserFS/releases/tag/v0.6.1); the bits should be identical to what I sent you.
